src/gui/properties/proplistdelegate.cpp

```diff
@@ -87,6 +87,13 @@ QWidget *PropListDelegate::createEditor(QWidget *parent, const QStyleOptionViewI
     editor->addItem(tr("Normal", "Normal (priority)"));
     editor->addItem(tr("High", "High (priority)"));
     editor->addItem(tr("Maximum", "Maximum (priority)"));
+
+    // Apply newly selected priority right away, don't wait until editor is closed.
+    connect(editor, qOverload<int>(&QComboBox::activated), this, [this, editor]()
```

Comment:
what about using `QComboBox::currentIndexChanged`?

Reply:
It seems to me that `activated` is better for the purpose - it's emitted in response to user interaction only while `currentIndexChanged` is emitted also when the index is being set programmatically.

Reply:
I'm concerned about this part:
https://doc.qt.io/qt-5/qcombobox.html#activated
>Note that this signal is sent even when the choice is not changed. If you need to know when the choice actually changes, use signal currentIndexChanged() or currentTextChanged().

I'm under impression that you'll invoke `emit` unnecessarily with `QComboBox::activated`.

Reply:
Yes, you're right - this will happen when i.e. priority is set to "high" and user selects "high" again. However, just hooking to `currentIndexChanged` instead of `activated` would result in unnecessary lambda call during initialization of the combo box, that's worse for sure. Some additional code would have to be added i.e. hook to  `currentIndexChanged` but after initial value is set. I think that something else could be done. Use `activated` but check if the value really changed (compare against the model) inside`PropListDelegate::setModelData`
```diff
+    if (index.data(TorrentContentModel::UnderlyingDataRole).toInt() == priority)
+        return;
+
     model->setData(index, static_cast<int>(prio));
     emit filteredFilesChanged();
 }
```
This way is universal - it covers all cases where data is about to be sent to the model, not only those caused by the lambda. `PropListDelegate::setModelData` is triggered not only by the lambda but also automatically when the "editor" is being closed (when combo box looses focus). When "editor" is being closed, data doesn't have to be sent to the model because we already sent it earlier.